tests/TestBaseClass.php

```diff
@@ -238,4 +256,22 @@ protected static function addUserPermissions(int $userId, array $permissions)
         }
         return $results;
     }
+
+    /**
+     * Invokes a private method from an object
+     * Expected to be used for unit testing purposes on edge cases.
+     *
+     * @param object $object
+     * @param string $methodName
+     * @param array $parameters
+     * @return mixed
+     */
+    public function invokePrivateMethod(&$object, $methodName, array $parameters = array())
+    {
+        $reflection = new \ReflectionClass(get_class($object));
```

Comment:
Ah this is pretty nice. We've started to use another mocking framework too to get around PHPUnit limitations.

Reply:
Just a confirmation : this can not be used via **URL** hacking to call a private method and create security issue ?

Reply:
I can change it to protected probably.

Reply:
I really think it can not happen, only controller can be call directly. But why not protected, think it must be used only in test.

If needed elsewhere : move the function to public is better.

:+1:

Reply:
At last, I removed it. I think I initially used it and then just used the events.
When needed can be restablished